src/Idris/Desugar.idr

```diff
@@ -209,21 +209,22 @@ mutual
   desugarB side ps (PLam fc rig p pat argTy scope)
       = pure $ ILam EmptyFC rig !(traverse (desugar AnyExpr ps) p)
                    (Just (MN "lamc" 0)) !(desugarB AnyExpr ps argTy) $
-                 ICase fc (IVar EmptyFC (MN "lamc" 0)) (Implicit fc False)
+                 ICase fc top (IVar EmptyFC (MN "lamc" 0)) (Implicit fc False)
                      [snd !(desugarClause ps True (MkPatClause fc pat scope []))]
   desugarB side ps (PLet fc rig (PRef prefFC n) nTy nVal scope [])
```

Comment:
Here you should probably add a `if (isPatternVariable nm)` like in the `PLam` case on line 191.

Reply:
This change ended up breaking the test `idris2/real002`, which was doing:

```
     32 | storeProg : Has [Console, StoreI] e => App e ()
     33 | storeProg
     34 |     = let (>>=) = bindL in
                    ^^^^^
```

That looks like something people would want to do, so I'm also checking for `isOpName`.

Reply:
Oh, but Data.Fin.Extra is also broken by this change.  This function is easily repaired, but the `let` change may break other code. Do we still want to do it?
```
Error:While processing right hand side of invFinSpec. Undefined name H.
Data.Fin.Extra:57:25--57:26
53 |
54 | export
55 | invFinSpec : {n : _} -> (i : Fin n) -> 1 + finToNat i + finToNat (invFin i) = n
56 | invFinSpec {n = S k} FZ = cong S finToNatLastIsBound
57 | invFinSpec (FS k) = let H = invFinSpec k in
```

Reply:
I think we do want to do it, so that we're consistent about considering that upper case letters are constructor patterns (cf. LHS).

Reply:
The original change for the `PLam` case replaced `side` with `AnyExpr` in all of the recursive calls. I wasn't sure why. Should that be done in the `PLet` case? Does it have an effect aside from allowing `?` - I couldn't find any other code that matched or compared LHS / AnyExpr.